gui/src/main/java/org/verapdf/gui/ValidateWorker.java

```diff
@@ -64,149 +43,55 @@
-        File tempFile = File.createTempFile("fixedTempFile", ".pdf");
-        tempFile.deleteOnExit();
-        try (OutputStream tempOutput = new BufferedOutputStream(
-                new FileOutputStream(tempFile))) {
-            MetadataFixerResult fixerResult = MetadataFixerImpl.fixMetadata(
-                    tempOutput, fixerConfig);
-            MetadataFixerResult.RepairStatus repairStatus = fixerResult
-                    .getRepairStatus();
-            if (repairStatus == RepairStatus.SUCCESS || repairStatus == RepairStatus.ID_REMOVED) {
-                File resFile;
-                boolean flag = true;
-                while (flag) {
-                    if (!path.toString().trim().isEmpty()) {
-                        resFile = FileGenerator.createOutputFile(this.settings
-                                .getFixMetadataPathFolder().toFile(), this.pdf
-                                .getName(), this.settings
-                                .getMetadataFixerPrefix());
-                    } else {
-                        resFile = FileGenerator.createOutputFile(this.pdf,
-                                this.settings.getMetadataFixerPrefix());
-                    }
-
-                    try {
-                        Files.copy(tempFile.toPath(), resFile.toPath());
-                        flag = false;
-                    } catch (FileAlreadyExistsException e) {
-                        LOGGER.error(e);
-                    }
-                }
-            }
-            return fixerResult;
-        }
-    }
-
-    private ValidationResult runValidator(ModelParser toValidate)
-            throws IOException {
-        try {
-            int max = this.settings.getMaxNumberOfFailedChecks();
-            PDFAValidator validator;
-            if (max > 0) {
-                validator = Validators.createValidator(this.profile, true, max);
-            } else {
-                validator = Validators.createValidator(this.profile, true);
-            }
-            return validator.validate(toValidate);
-        } catch (ValidationException e) {
-
-            this.parent.errorInValidatingOccur(
-                    GUIConstants.ERROR_IN_VALIDATING, e);
-        }
-        return null;
+    protected Void doInBackground() {
+		try {
+			this.xmlReport = File.createTempFile("veraPDF-tempXMLReport", ".xml");
+			this.xmlReport.deleteOnExit();
+			this.htmlReport = null;
+		} catch (IOException e) {
+			LOGGER.error("Can't create temporary file for XML report", e);
+		}
+		try (InputStream toProcess = new FileInputStream(pdf);
+		OutputStream mrrReport = new FileOutputStream(this.xmlReport)) {
+			Processor processor = new ProcessorImpl();
+			processingResult = processor.validate(toProcess, ItemDetails.fromFile(pdf),
+					settings, mrrReport);
+		} catch (IOException e) {
+			LOGGER.error("Can't open stream from PDF file or can't open stream to temporary XML report file", e);
+		}
+
+        writeHtmlReport();
```

Comment:
This method call should be in above try block. We shouldn't call it if we did not obtain an xml report